### PR TITLE
Feature/1836 min length ids v2

### PR DESCRIPTION
**Pull request by crbrox**
Implements #1836

@fgalan 
@kzangeli

### CODE REVIEW

File: test/functionalTest/cases/1836_min_length_id_v2/1836_min_length_id_v2_metadata.test [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Min lenght for IDs in V2 (metadata)
```
Comment (kzangeli) [inline test/functionalTest/cases/1836_min_length_id_v2/1836_min_length_id_v2_metadata.test]: Typo: length

File: src/lib/jsonParseV2/parseAttributeValue.cpp [hunk 2]
```diff
@@ -68,5 +68,9 @@ std::string parseAttributeValue(ConnectionInfo* ciP, ContextAttribute* caP)
   caP->valueType  = (document.IsObject())? orion::ValueTypeObject : orion::ValueTypeVector;
   parseContextAttributeCompoundValueStandAlone(document, caP, caP->valueType);
 
+  if (!caP->typeGiven) {
```
Comment (kzangeli) [inline src/lib/jsonParseV2/parseAttributeValue.cpp]: { on separate line

File: test/functionalTest/cases/1259_new_json_for_v2/update_attribute_with_null_value.test [hunk 3]
```diff
@@ -34,13 +34,13 @@ brokerStart CB 0
...
-# 05. Get E2/A2 (A2 stays the same)
+# 05. Get E2/A2 (value of A2 has been replaced)
 # 06. Create E3/A3='06'
 # 07. Modify E3/A3=null (PUT)
 # 08. Get E3/A3 (value of A3 has been replaced)
 # 09. Create E4/A4='09'
 # 10. Modify E4/A4=null (POST)
-# 11. Get E4/A4 (A4 stays the same)
+# 11. Get E4/A4 (value of A4 has been replaced)
 # 12. Create (POST /v2/entities) E5/A5=null using URI param keyValues
```
Comment (kzangeli) [inline test/functionalTest/cases/1259_new_json_for_v2/update_attribute_with_null_value.test]: This here gives me the impression a bug fix has been done ...
Or is it just a fix of an erroneous functest of an older bugfix?

File: test/functionalTest/cases/1836_min_length_id_v2/1836_min_length_id_v2_metadata.test (hunk 1 above)
Comment (crbrox) [inline test/functionalTest/cases/1836_min_length_id_v2/1836_min_length_id_v2_metadata.test]: Fixed bdc93f6

File: src/lib/jsonParseV2/parseAttributeValue.cpp (hunk 2 above)
Comment (crbrox) [inline src/lib/jsonParseV2/parseAttributeValue.cpp]: fixed 2c239d9

File: test/functionalTest/cases/1259_new_json_for_v2/update_attribute_with_null_value.test (hunk 3 above)
Comment (fgalan) [inline test/functionalTest/cases/1259_new_json_for_v2/update_attribute_with_null_value.test]: Yes, it was a bug, fixed in this PR as a side-effect.

**Comment by fgalan** [comment]
LGTM (althought I'm not sure if @kzangeli's comment in https://github.com/telefonicaid/fiware-orion/pull/1877#discussion_r54883186 could involve an action, maybe adding something to CNR about the bugfix?)

**Comment by kzangeli** [comment]
Yes, that was my thought, some line in CNR.

LGTM about the rest

File: test/functionalTest/cases/1259_new_json_for_v2/update_attribute_with_null_value.test (hunk 3 above)
Comment (crbrox) [inline test/functionalTest/cases/1259_new_json_for_v2/update_attribute_with_null_value.test]: Fixed 69cc2cf

**Pull request merged**